### PR TITLE
Allow ed25519 gas

### PR DESCRIPTION
**Pull request by hong-pang**
1. changed app.SetAnteHandler, removed "ante.DefaultSigVerificationGasConsumer", added "helpers.StSigVerificationGasConsumer"
2. in "x/register/msg.go", added "NodeSign"  
3. added test file to test pubKey derived from Ed25519. Test passed
4. in "x/register/app_test.go", introduce anteHandler := ante.NewAnteHandler(mApp.AccountKeeper, supplyKeeper, helpers.StSigVerificationGasConsumer) to simulate real customized "anteHandler".
5. moved "anteHelper.go" from app folder to helpers folder to avoid cycle importing

### CODE REVIEW

File: x/register/types/msg.go [hunk 1]
```diff
@@ -74,6 +76,7 @@ func (msg MsgCreateResourceNode) GetSignBytes() []byte {
 
 func (msg MsgCreateResourceNode) GetSigners() []sdk.AccAddress {
 	// OwnerAddress is first signer so Owner pays fees
+	// and since the second signer
```
Comment (alexstratos) [inline x/register/types/msg.go:79]: finish this comments

**Review by alexstratos — APPROVED** [review]

File: x/register/allow-ed25519-gas_test.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
+package register
```
[the remark this answers is not in the document]
Comment (hong-pang) [reply x/register/allow-ed25519-gas_test.go:1]: changed the name with "_"

File: x/register/types/msg.go (hunk 1 above)
Comment (hong-pang) [reply x/register/types/msg.go]: updated

Comment (hong-pang) [reply x/register/types/msg.go]: completed the comments.

**Review by alexstratos — APPROVED** [review]

**Pull request merged**